src/devices/bus/spectrum/neogs.h

```diff
@@ -6,36 +6,23 @@
 
 #pragma once
 
+#include "zxbus.h"
+
 #include "cpu/z80/z80.h"
 #include "machine/ram.h"
 #include "sound/dac.h"
 
-DECLARE_DEVICE_TYPE(NEOGS, neogs_device)
-
-class neogs_device : public device_t
+class neogs_device : public device_t, public device_zxbus_card_interface
```

Comment:
Since this device is only used via the `device_zxbus_card_interface`, you can move it to an anonymous namespace in the .cpp file to cut down on global symbols.  See e.g. src/devices/bus/gameboy, src/devices/bus/sms_ctrl or src/devices/bus/msx/ctrl for examples of how to have devices with private implementations.